modules/storage/mysql-single-server/modules/privateEndpoint.bicep

```diff
@@ -0,0 +1,58 @@
+param location string
+param tags object
+param privateEndpoints array
+
+var varPrivateEndpoints = [for (p, i) in privateEndpoints: {
+  name: p.name
+  privateLinkServiceId: p.privateLinkServiceId
+  groupIds: p.groupIds
+  subnetId: p.subnetId
+  privateDnsZones: contains(p, 'privateDnsZones') ? p.privateDnsZones : []
+  customNetworkInterfaceName: contains(p, 'customNetworkInterfaceName') ? p.customNetworkInterfaceName : null
+  manualApprovalEnabled: contains(p, 'manualApprovalEnabled') ? p.manualApprovalEnabled : false
+}]
```

Comment:
The parameter already appears to be in this format when it is based into the module. Since this module is internal, I don't think you have to recreate the parameter.

Optionally, you could leverage a user defined type here if you would like to try it out. It is in preview, but we have enabled the feature for the repo. You would just need to add a bicepconfig.json into the directory like [this](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/user-defined-data-types#enable-the-preview-feature).

Reply:
```suggestion
param privateEndpoints array
```